102-bridge-vertx5-errors.md

```diff
@@ -0,0 +1,195 @@
+# Additional OpenAPI schema validation error details with Vert.x 5
+
+This proposal outlines an enhancement to the Strimzi HTTP Bridge error responses, adding detailed information for OpenAPI schema validation failures.
+The change is prompted by updates introduced in Vert.x as part of the upcoming upgrade from version 4.x to 5.x.
+
+## Current situation
+
+The Strimzi HTTP bridge uses Vert.x 4.x, consistent with all Strimzi operators.
+By leveraging the `vertx-web-openapi` component, together with the underneath JSON schema validation dependency, the bridge can validate incoming HTTP requests against the OpenAPI v3 specification before processing them.
+When a schema validation error occurs, the HTTP response body returned to the client has the following format:
+
+```json
+{
+    "error_code": 400,
+    "message": "<description of the schema validation error>"
+}
+```
+
+The content of the `message` field is derived from the `ValidationException` raised during JSON schema validation by the `vertx-web-openapi` component and is caught by the bridge while handling the request.
+
+For example, if an HTTP client attempts to create a new Kafka consumer but provides the `enable.auto.commit` field as a string value `"true"` instead of the expected boolean `true`, the schema validation will fail, and the following error will be returned:
+
+```json
+{
+    "error_code": 400,
+    "message": "Validation error on: /enable.auto.commit - input don't match type BOOLEAN"
```

Comment:
```suggestion
    "message": "Validation error on: /enable.auto.commit - input doesn't match type BOOLEAN"
```

Reply:
I can't change this. It's the actual message we get from the Vert.x JSON schema library :-)

Reply:
very easy contribution to the Vert.x I would say :D.